### PR TITLE
Sampling speed up

### PR DESCRIPTION
**Pull request by WalterSimoncini**
- Sped up the sampling process by removing `torch.where` calls and vectorizing loops in `get_class_frequency` and `get_patch_scores `
- Added an `eval.py` script to quicky run experiments
- Minor updated in `README.md` to show how to use the `eval.py` script and quickly download the data via shell

### CODE REVIEW

**Comment by WalterSimoncini** [comment]
Sampling was sped up by 7-10x (and up to 30x in some cases). Replicated the result with a DINO ViT 16/S using a memory bank of 1024 * 10^2 and obtained a mIoU of 37.12 (the difference might be due to seed differences).
